test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJmap.java

```diff
@@ -0,0 +1,174 @@
+		OpenJ9VirtualMachine myVm = (OpenJ9VirtualMachine) vm;
+		InputStream result = myVm.heapHisto();
+		checkForProperties(result);
+		result = myVm.heapHisto("-live"); //$NON-NLS-1$
+		checkForProperties(result);
+		result = myVm.heapHisto("-all"); //$NON-NLS-1$
+		checkForProperties(result);
+	}
+
+	@BeforeSuite
+	protected void setupSuite() {
+		assertTrue("This test is valid only on OpenJ9", //$NON-NLS-1$
+				System.getProperty("java.vm.vendor").contains("OpenJ9")); //$NON-NLS-1$//$NON-NLS-2$
+		getJdkUtilityPath(JMAP_COMMAND);
+		myId = TargetManager.getVmId();
+		assertNotNull(myId, "Attach ID missing"); //$NON-NLS-1$
+		log(myId);
+		
+		/* make test objects */
+		hws = new ArrayList<>();
+		for (int i = 0; i < 3; i++) {
+			hws.add(new HelloWorld());
+		}
+		objectArray = new String[][] {{"a", "b"}, {"c", "d"}};   //$NON-NLS-1$//$NON-NLS-2$ //$NON-NLS-3$//$NON-NLS-4$
+		objectVector = new Integer[] {Integer.valueOf(10), Integer.valueOf(20)};
+		primitiveArray = new int[][] {{1, 2}, {3, 4}};
+		primitiveVector = new long[] {5, 6};
+	}
+
+	private void checkHeapStats(List<String> heapStats) {
+		String hwClassName = HelloWorld.class.getName();
+		String expectedClasses[] = new String[] {hwClassName, 
+				primitiveArray.getClass().getName(),
+				primitiveVector.getClass().getName(),
+				objectArray.getClass().getName(),
+				objectVector.getClass().getName(),
+		};
+		for (String className: expectedClasses) {
+			Optional<String> result = searchSubstring(className, heapStats);
+			assertTrue(result.isPresent(), "stats missing for " + className); //$NON-NLS-1$
+		}
+		Optional<String> result = searchSubstring(hwClassName, heapStats);
+		String stringResult = result.get();
+		assertTrue(stringResult.matches(".*\\b3\\b.*"), "wrong number of HelloWorld objects: "+stringResult);  //$NON-NLS-1$//$NON-NLS-2$
+	}
+
+	private void checkForProperties(InputStream result) throws IOException {
+		try (BufferedReader jpsOutReader = new BufferedReader(new InputStreamReader(result))) {
+			List<String> outputLines = jpsOutReader.lines().collect(Collectors.toList());
+			String propertiesName = Properties.class.getName();
+			assertTrue(searchSubstring(propertiesName, outputLines).isPresent(), "Missing class: " + propertiesName); //$NON-NLS-1$
+		}
+	}
+
+	@AfterTest
+	private void preserveHelloWorld() {
+		/* force the objects to stay alive */
+		for (HelloWorld instance: hws) {
+			log(instance.getClass().getName());
+		}
+		log(Integer.toString(objectArray.hashCode()));
+		log(Integer.toString(objectVector.hashCode()));
+		log(Integer.toString(primitiveArray.hashCode()));
+		log(Integer.toString(primitiveVector.hashCode()));
+	}
+
+	private void logJmapOutput(List<String> jmapOutput) {
+		StringBuilder buff = new StringBuilder("jmap output:\n"); //$NON-NLS-1$
+		jmapOutput.forEach(s -> {
+			buff.append(s);
+			buff.append("\n"); //$NON-NLS-1$
+		});
+		log(buff.toString());
+	}
+
+	static class Finalizable {
```

Comment:
Is this used?

Reply:
At the moment no; it is a legacy from the finalizerinfo feature.